tests/test_repository_pypi.py

```diff
@@ -126,3 +128,50 @@ def test_open_local_or_remote_file__remote_file(
             assert file_stream.size == expected_content_length
 
     mock_response.close.assert_called_once()
+
+
+def test_pypirepo_build_dir_is_str(pypi_repository):
+    assert isinstance(pypi_repository.build_dir, str)
+
+
+def test_pypirepo_source_dir_is_str(pypi_repository):
+    assert isinstance(pypi_repository.source_dir, str)
+
+
+@pytest.mark.skipif(
+    PIP_VERSION >= (10,),
+    reason="RequirementSet objects don't take arguments after pip 10.",
+)
+def test_pypirepo_calls_reqset_with_str_paths(pypi_repository):
+    """
+    Make sure that paths passed to RequirementSet init are str.
+
+    Passing unicode paths on Python 2 could make pip fail later on
+    unpack, if the package contains non-ASCII file names, because
+    non-ASCII str and unicode paths cannot be combined.
+    """
+    with patch("piptools.repositories.pypi.RequirementSet") as mocked_init:
```

Comment:
```suggestion
    with mock.patch("piptools.repositories.pypi.RequirementSet") as mocked_init:
```

Use imported `mock`.